File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/KbVHnswIndexDetails.java
package com.oracle.aidataplatform.dp.model;



/**
 * HNSW (Hierarchical Navigable Small World) index parameters
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=KbVHnswIndexDetails.Builder.class)

public final class KbVHnswIndexDetails  {
    @Deprecated
    @java.beans.ConstructorProperties({"distance", "targetAccuracy", "neighbors", "efConstruction"})
    public KbVHnswIndexDetails(Distance distance, Integer targetAccuracy, Integer neighbors, Integer efConstruction) {
        super();
        this.distance = distance;
        this.targetAccuracy = targetAccuracy;
        this.neighbors = neighbors;
        this.efConstruction = efConstruction;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Distance metric for the vector index
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("distance")
private Distance distance;

        /**
         * Distance metric for the vector index
         * @param distance the value to set
         * @return this builder
         **/
        

public Builder distance(Distance distance) {
    this.distance = distance;
    return this;
}
            /**
     * Target accuracy percentage for the index (1-100)
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("targetAccuracy")
private Integer targetAccuracy;

        /**
         * Target accuracy percentage for the index (1-100)
         * @param targetAccuracy the value to set
         * @return this builder
         **/
        

public Builder targetAccuracy(Integer targetAccuracy) {
    this.targetAccuracy = targetAccuracy;
    return this;
}
            /**
     * Maximum number of neighbors each vector can have on any layer (M parameter)
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("neighbors")
private Integer neighbors;

        /**
         * Maximum number of neighbors each vector can have on any layer (M parameter)
         * @param neighbors the value to set
         * @return this builder
         **/
        

public Builder neighbors(Integer neighbors) {
    this.neighbors = neighbors;
    return this;
}
            /**
     * Maximum number of closest vector candidates considered during index construction
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("efConstruction")
private Integer efConstruction;

        /**
         * Maximum number of closest vector candidates considered during index construction
         * @param efConstruction the value to set
         * @return this builder
         **/
        

public Builder efConstruction(Integer efConstruction) {
    this.efConstruction = efConstruction;
    return this;
}


        public KbVHnswIndexDetails build() {
            KbVHnswIndexDetails model = new KbVHnswIndexDetails(this.distance
                , this.targetAccuracy
                , this.neighbors
                , this.efConstruction);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(KbVHnswIndexDetails model) {
                this.distance(model.getDistance());
    this.targetAccuracy(model.getTargetAccuracy());
    this.neighbors(model.getNeighbors());
    this.efConstruction(model.getEfConstruction());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    

    /**
     * Distance metric for the vector index
     **/
    public enum Distance implements com.oracle.bmc.http.internal.BmcEnum {
        Cosine("COSINE"),
        Euclidean("EUCLIDEAN"),
        L2Squared("L2_SQUARED"),
        Dot("DOT"),
        Manhattan("MANHATTAN"),
        Hamming("HAMMING"),
        Jaccard("JACCARD"),
        ;

        

        private final String value;
        private static java.util.Map<String, Distance> map;

        static {
            map = new java.util.HashMap<>();
            for (Distance v : Distance.values()) {
                    map.put(v.getValue(), v);
                
            }
        }

        Distance(String value) {
            this.value = value;
        }

        @com.fasterxml.jackson.annotation.JsonValue
        public String getValue() {
            return value;
        }

        @com.fasterxml.jackson.annotation.JsonCreator
        public static Distance create(String key) {
            if (map.containsKey(key)) {
                return map.get(key);
            }
            throw new IllegalArgumentException("Invalid Distance: " + key);
        }
    };
        /**
     * Distance metric for the vector index
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("distance")
    private final Distance distance;

        /**
     * Distance metric for the vector index
     * @return the value
     **/
    
    public Distance getDistance() {
        return distance;
    }


        /**
     * Target accuracy percentage for the index (1-100)
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("targetAccuracy")
    private final Integer targetAccuracy;

        /**
     * Target accuracy percentage for the index (1-100)
     * @return the value
     **/
    
    public Integer getTargetAccuracy() {
        return targetAccuracy;
    }


        /**
     * Maximum number of neighbors each vector can have on any layer (M parameter)
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("neighbors")
    private final Integer neighbors;

        /**
     * Maximum number of neighbors each vector can have on any layer (M parameter)
     * @return the value
     **/
    
    public Integer getNeighbors() {
        return neighbors;
    }


        /**
     * Maximum number of closest vector candidates considered during index construction
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("efConstruction")
    private final Integer efConstruction;

        /**
     * Maximum number of closest vector candidates considered during index construction
     * @return the value
     **/
    
    public Integer getEfConstruction() {
        return efConstruction;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("KbVHnswIndexDetails(");
        sb.append("distance=").append(String.valueOf(this.distance));
        sb.append(", targetAccuracy=").append(String.valueOf(this.targetAccuracy));
        sb.append(", neighbors=").append(String.valueOf(this.neighbors));
        sb.append(", efConstruction=").append(String.valueOf(this.efConstruction));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof KbVHnswIndexDetails)) {
            return false;
        }

        KbVHnswIndexDetails other = (KbVHnswIndexDetails) o;
        return java.util.Objects.equals(this.distance, other.distance) &&
            java.util.Objects.equals(this.targetAccuracy, other.targetAccuracy) &&
            java.util.Objects.equals(this.neighbors, other.neighbors) &&
            java.util.Objects.equals(this.efConstruction, other.efConstruction);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.distance == null ? 43 : this.distance.hashCode());
        result = (result * PRIME) + (this.targetAccuracy == null ? 43 : this.targetAccuracy.hashCode());
        result = (result * PRIME) + (this.neighbors == null ? 43 : this.neighbors.hashCode());
        result = (result * PRIME) + (this.efConstruction == null ? 43 : this.efConstruction.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/kb_v_hnsw_index_details.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class KbVHnswIndexDetails(object):
    """
    HNSW (Hierarchical Navigable Small World) index parameters
    """

    #: A constant which can be used with the distance property of a KbVHnswIndexDetails.
    #: This constant has a value of "COSINE"
    DISTANCE_COSINE = "COSINE"

    #: A constant which can be used with the distance property of a KbVHnswIndexDetails.
    #: This constant has a value of "EUCLIDEAN"
    DISTANCE_EUCLIDEAN = "EUCLIDEAN"

    #: A constant which can be used with the distance property of a KbVHnswIndexDetails.
    #: This constant has a value of "L2_SQUARED"
    DISTANCE_L2_SQUARED = "L2_SQUARED"

    #: A constant which can be used with the distance property of a KbVHnswIndexDetails.
    #: This constant has a value of "DOT"
    DISTANCE_DOT = "DOT"

    #: A constant which can be used with the distance property of a KbVHnswIndexDetails.
    #: This constant has a value of "MANHATTAN"
    DISTANCE_MANHATTAN = "MANHATTAN"

    #: A constant which can be used with the distance property of a KbVHnswIndexDetails.
    #: This constant has a value of "HAMMING"
    DISTANCE_HAMMING = "HAMMING"

    #: A constant which can be used with the distance property of a KbVHnswIndexDetails.
    #: This constant has a value of "JACCARD"
    DISTANCE_JACCARD = "JACCARD"

    def __init__(self, **kwargs):
        """
        Initializes a new KbVHnswIndexDetails object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param distance:
            The value to assign to the distance property of this KbVHnswIndexDetails.
            Allowed values for this property are: "COSINE", "EUCLIDEAN", "L2_SQUARED", "DOT", "MANHATTAN", "HAMMING", "JACCARD"
        :type distance: str

        :param target_accuracy:
            The value to assign to the target_accuracy property of this KbVHnswIndexDetails.
        :type target_accuracy: int

        :param neighbors:
            The value to assign to the neighbors property of this KbVHnswIndexDetails.
        :type neighbors: int

        :param ef_construction:
            The value to assign to the ef_construction property of this KbVHnswIndexDetails.
        :type ef_construction: int

        """
        self.swagger_types = {
            'distance': 'str',
            'target_accuracy': 'int',
            'neighbors': 'int',
            'ef_construction': 'int'
        }

        self.attribute_map = {
            'distance': 'distance',
            'target_accuracy': 'targetAccuracy',
            'neighbors': 'neighbors',
            'ef_construction': 'efConstruction'
        }

        self._distance = None
        self._target_accuracy = None
        self._neighbors = None
        self._ef_construction = None

    @property
    def distance(self):
        """
        Gets the distance of this KbVHnswIndexDetails.
        Distance metric for the vector index

        Allowed values for this property are: "COSINE", "EUCLIDEAN", "L2_SQUARED", "DOT", "MANHATTAN", "HAMMING", "JACCARD"


        :return: The distance of this KbVHnswIndexDetails.
        :rtype: str
        """
        return self._distance

    @distance.setter
    def distance(self, distance):
        """
        Sets the distance of this KbVHnswIndexDetails.
        Distance metric for the vector index


        :param distance: The distance of this KbVHnswIndexDetails.
        :type: str
        """
        allowed_values = ["COSINE", "EUCLIDEAN", "L2_SQUARED", "DOT", "MANHATTAN", "HAMMING", "JACCARD"]
        if not value_allowed_none_or_none_sentinel(distance, allowed_values):
            raise ValueError(
                "Invalid value for `distance`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._distance = distance

    @property
    def target_accuracy(self):
        """
        Gets the target_accuracy of this KbVHnswIndexDetails.
        Target accuracy percentage for the index (1-100)


        :return: The target_accuracy of this KbVHnswIndexDetails.
        :rtype: int
        """
        return self._target_accuracy

    @target_accuracy.setter
    def target_accuracy(self, target_accuracy):
        """
        Sets the target_accuracy of this KbVHnswIndexDetails.
        Target accuracy percentage for the index (1-100)


        :param target_accuracy: The target_accuracy of this KbVHnswIndexDetails.
        :type: int
        """
        self._target_accuracy = target_accuracy

    @property
    def neighbors(self):
        """
        Gets the neighbors of this KbVHnswIndexDetails.
        Maximum number of neighbors each vector can have on any layer (M parameter)


        :return: The neighbors of this KbVHnswIndexDetails.
        :rtype: int
        """
        return self._neighbors

    @neighbors.setter
    def neighbors(self, neighbors):
        """
        Sets the neighbors of this KbVHnswIndexDetails.
        Maximum number of neighbors each vector can have on any layer (M parameter)


        :param neighbors: The neighbors of this KbVHnswIndexDetails.
        :type: int
        """
        self._neighbors = neighbors

    @property
    def ef_construction(self):
        """
        Gets the ef_construction of this KbVHnswIndexDetails.
        Maximum number of closest vector candidates considered during index construction


        :return: The ef_construction of this KbVHnswIndexDetails.
        :rtype: int
        """
        return self._ef_construction

    @ef_construction.setter
    def ef_construction(self, ef_construction):
        """
        Sets the ef_construction of this KbVHnswIndexDetails.
        Maximum number of closest vector candidates considered during index construction


        :param ef_construction: The ef_construction of this KbVHnswIndexDetails.
        :type: int
        """
        self._ef_construction = ef_construction

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
